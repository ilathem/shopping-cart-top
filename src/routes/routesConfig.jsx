import PageNotFound from './page-not-found';
import Root from './root';
import Home from './home';
import Shop from './shop';

const routesConfig = [
  {
    path: '/',
    element: <Root />,
    children: [
      {
        index: true,
        element: <Home />,
      },
      {
        path: '/shop',
        element: <Shop />,
      },
      {
        path: '/*',
        element: <PageNotFound />,
      },
    ],
  },
];

export default routesConfig;
