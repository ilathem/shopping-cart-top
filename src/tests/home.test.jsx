
import { describe, it, expect, beforeEach } from 'vitest';
import { render, screen } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router';
import routesConfig from '../routes/routesConfig';
import Home from '../routes/home';

beforeEach(() => { // Arrange in context of route component 
  const router = createMemoryRouter(routesConfig, {
    initialEntries: ['/'],
  });
  render(<RouterProvider router={router} />);
});

describe('Home component', () => {
  it('renders correct heading', () => {
    expect(screen.getByRole('heading', { name: 'page-title' }).textContent).toMatch(/Home/); // Assert
  })
  
  it('displays greeting component', () => {
    expect(screen.getByText(/Welcome/));
  })

  it('displays author', () => {
    expect(screen.getByText(/Isaiah Lathem/));
  })

  it('has a link to the repo', () => {
    expect(screen.getByRole('link', {name: /repo/i})).toHaveAttribute('href', 'https://github.com/ilathem/shopping-cart-top');
  } )
  
  it('renders snapshot correctly', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot();
  })
})