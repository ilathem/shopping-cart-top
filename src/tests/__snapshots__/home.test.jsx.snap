// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > renders snapshot correctly 1`] = `
<div>
  <div
    class="m-5 border-2 p-4 rounded-2xl ml-0"
  >
    <h1
      class="text-4xl border-b-3 border-zinc-600 pb-1 mb-2"
    >
      Home page
    </h1>
    <div>
      <section
        aria-label="introduction"
        class="p-2 border-2 border-zinc-700/50 my-2 rounded-xl hover:border-zinc-700 transition-all"
      >
        <h2
          aria-label="welcome-text"
          class="text-2xl"
        >
          Welcome to the shopping cart app!
        </h2>
        <p
          aria-label="creator"
        >
          Created by Isaiah Lathem
        </p>
      </section>
      <section
        aria-label="how-to-use"
        class="p-2 my-2 border-2 rounded-xl border-zinc-700/50 hover:border-zinc-700 transition-all"
      >
        <h2>
          How to use:
        </h2>
        <p>
          Click "Shop" in the Nav to navigate to the shop page, then add various items to the cart by updating their quantity and clicking Add to Cart.
        </p>
      </section>
      <section
        aria-label="repository link"
        class="p-2 my-2 border-2 rounded-xl border-zinc-700/50 hover:border-zinc-700 transition-all"
      >
        <h2>
          <a
            class="hover:text-zinc-200 transition-all"
            href="https://github.com/ilathem/shopping-cart-top"
            target="_blank"
          >
            Click here to visit the GitHub repo
          </a>
        </h2>
      </section>
    </div>
  </div>
</div>
`;
